src/rdkafka.h

```diff
@@ -204,7 +204,18 @@ typedef enum rd_kafka_timestamp_type_t {
 	RD_KAFKA_TIMESTAMP_LOG_APPEND_TIME  /**< Log append time */
 } rd_kafka_timestamp_type_t;
 
-
+/**
+ * @brief Set external allocator for rd_malloc/rd_calloc/rd_realloc/rd_free.
```

Comment:
rd_malloc, et.al, are unknown to the users of librdkafka, rephrase as:
`Set external allocator for allocations made by librdkafka`

Also make a list of methods that are affected by this allocator, things like RD_KAFKA_MSG_F_FREE to produce(), rd_kafka_free(), etc.